src/Agent.Worker/TestResults/ResultsCommands.cs

```diff
@@ -112,108 +106,126 @@ private void PublishTestResults(IResultReader resultReader)
             }
         }
 
         /// <summary>
         /// Publish separate test run for each result file that has results.
         /// </summary>
-        private async Task PublishToNewTestRunPerTestResultFileAsync(List<string> resultFiles, ITestRunPublisher publisher)
+        private async Task PublishToNewTestRunPerTestResultFileAsync(List<string> resultFiles, ITestRunPublisher publisher, string resultReader)
         {
-            int runCount = 1;
-            // Publish separate test run for each result file that has results.
-            foreach (string resultFile in resultFiles)
+            try
             {
-                string runName = null;
-                if (!string.IsNullOrWhiteSpace(_runTitle))
+                int runCount = 1;
+                // Publish separate test run for each result file that has results.
+                foreach (string resultFile in resultFiles)
                 {
-                    runName = resultFiles.Count > 1 ? string.Format(CultureInfo.InvariantCulture, "{0} {1}", _runTitle, runCount++) : _runTitle;
-                }
+                    string runName = null;
+                    if (!string.IsNullOrWhiteSpace(_runTitle))
+                    {
+                        runName = resultFiles.Count > 1 ? string.Format(CultureInfo.InvariantCulture, "{0} {1}", _runTitle, runCount++) : _runTitle;
+                    }
 
-                TestRunData testRunData = publisher.ReadResultsFromFile(resultFile, runName);
+                    _executionContext.Debug(StringUtil.Format("Reading test results from file '{0}'", resultFile));
+                    TestRunData testRunData = publisher.ReadResultsFromFile(resultFile, runName);
 
-                if (testRunData != null && testRunData.Results != null && testRunData.Results.Length > 0)
-                {
-                    publisher.StartTestRunAsync(testRunData).Wait();
-                    publisher.AddResultsAsync(testRunData.Results).Wait();
-                    await publisher.EndTestRunAsync();
-                }
-                else
-                {
-                    _executionContext.Warning(StringUtil.Loc("InvalidResultFiles", resultFile));
+                    if (testRunData != null && testRunData.Results != null && testRunData.Results.Length > 0)
+                    {
+                        publisher.StartTestRunAsync(testRunData).Wait();
+                        publisher.AddResultsAsync(testRunData.Results).Wait();
+                        await publisher.EndTestRunAsync();
+                    }
+                    else
+                    {
+                        _executionContext.Warning(StringUtil.Loc("InvalidResultFiles", resultFile, resultReader));
+                    }
                 }
             }
+            catch (Exception ex)
+            {
+                //Do not fail the task.
+                LogPublishTestResultsFailureWarning(ex);
```

Comment:
do we fail the task is all resultfiles weren't published. is the behaviour same as old agent?